app/services/concerns/xccdf/rule_results.rb

```diff
@@ -4,15 +4,25 @@ module Xccdf
   # Methods related to saving RuleResults from openscap_parser
   module RuleResults
     def save_rule_results
-      @rule_results = selected_op_rule_results.map do |op_rule_result|
+      ::RuleResult.import!(rule_results.select(&:new_record?), ignore: true)
+    end
+
+    def rule_results
+      @rule_results ||= selected_op_rule_results.map do |op_rule_result|
```

Comment:
Is this called more often within the process of saving rule results? It might not be necessary to memoize this or even store it in an instance variable.

Reply:
Yes. See `def failed_rules` below.